src/components/Player/Player.tsx

```diff
@@ -0,0 +1,157 @@
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+
+import styles from './Player.module.scss';
+
+import { addScript } from '#src/utils/dom';
+import { deepCopy } from '#src/utils/collection';
+import type { JWPlayer } from '#types/jwplayer';
+import type { PlaylistItem } from '#types/playlist';
+import useEventCallback from '#src/hooks/useEventCallback';
+import useOttAnalytics from '#src/hooks/useOttAnalytics';
+import { logDev } from '#src/utils/common';
+
+type Props = {
+  playerId?: string | null;
+  feedId?: string;
+  item: PlaylistItem;
+  onReady?: (player?: JWPlayer) => void;
+  onPlay?: () => void;
+  onPause?: () => void;
+  onComplete?: () => void;
+  onUserActive?: () => void;
+  onUserInActive?: () => void;
+  onBeforePlay?: () => void;
+  onPlaylistItemCallback?: (item: PlaylistItem) => Promise<undefined | PlaylistItem>;
+  startTime?: number;
+};
+
+const Player: React.FC<Props> = ({
+  playerId,
+  item,
+  onReady,
+  onPlay,
+  onPause,
+  onComplete,
+  onUserActive,
+  onUserInActive,
+  onBeforePlay,
+  onPlaylistItemCallback,
+  feedId,
+  startTime = 0,
+}: Props) => {
+  const playerElementRef = useRef<HTMLDivElement>(null);
+  const playerRef = useRef<JWPlayer>();
+  const loadingRef = useRef(false);
+  const [libLoaded, setLibLoaded] = useState(!!window.jwplayer);
+  const startTimeRef = useRef(startTime);
+  const scriptUrl = `${import.meta.env.APP_API_BASE_URL}/libraries/${playerId}.js`;
+  const setPlayer = useOttAnalytics(item, feedId);
+
+  const handleBeforePlay = useEventCallback(onBeforePlay);
+  const handlePlay = useEventCallback(onPlay);
+  const handlePause = useEventCallback(onPause);
+  const handleComplete = useEventCallback(onComplete);
+  const handleUserActive = useEventCallback(onUserActive);
+  const handleUserInactive = useEventCallback(onUserInActive);
+  const handlePlaylistItemCallback = useEventCallback(onPlaylistItemCallback);
+  const handleReady = useEventCallback(() => onReady && onReady(playerRef.current));
+
+  const attachEvents = useCallback(() => {
```

Comment:
Do playerRef or playerRef.current need to be a dependency to these 2 useCallback calls?

Reply:
Refs don't have an effect on the effect's lifecycle. Perhaps it's better for the `attachEvents` function to accept the playerInstance instead of relying on the ref so that we get typescript errors when we change this later.

The `attachEvents` and `detachEvents` functions are only used on setup and unmount.

Reply:
OK, what I was worried about is that this seems like a place we might get memory leaks if the playerRef changes and the events don't attach or detach properly, but it seems to be working ok.